server/channels/store/sqlstore/user_store.go

```diff
@@ -1674,7 +1674,20 @@ func (us SqlUserStore) performSearch(query sq.SelectBuilder, term string, option
 
 func (us SqlUserStore) AnalyticsGetInactiveUsersCount() (int64, error) {
 	var count int64
-	err := us.GetReplicaX().Get(&count, "SELECT COUNT(Id) FROM Users WHERE DeleteAt > 0")
+	query := us.getQueryBuilder().
+		Select("COUNT(Id)").
+		From("Users").
+		LeftJoin("Bots ON Users.ID = Bots.UserId").
+		Where(sq.And{
+			sq.Gt{"Users.DeleteAt": 0},
+			sq.Eq{"Bots.UserId": nil},
```

Comment:
Similar code in (https://github.com/mattermost/mattermost/blob/075681a412b56d177d54274f03852e4c93632f5a/server/channels/store/sqlstore/user_store.go#L1400) is using different query for postgres and mysql.  Perhaps the `Count` function can also be reused in order to keep logic in a single place?

Reply:
I've changed the code to address the performance differences between Mysql and Postgresql